NavigationFlowCoordinator/NavigationFlowCoordinator.swift

```diff
@@ -82,7 +75,16 @@ open class NavigationFlowCoordinator: FlowCoordinator {
     }
 
 
-    // MARK: Navigation controller related flow
+    // MARK: - Navigation controller related flow
+
+    func navigate(to viewController: UIViewController, with presentationStyle: navigationStyle, animated: Bool = true) {
+        switch presentationStyle {
+        case .present:
+            present(viewController: viewController, animated: animated)
```

Comment:
I like the idea of coordinator being able to to show its main VC modally, using .present presentationStyle. 

Unfortunately it won't work properly in the shape you proposed. Problems: 

1. NavigationControllerCoordinatorsTracker stores references to child coordinators in a map where mainViewControllers are mapped to its related coordinators. When cleaning references to coordianators we check if mainViewController is on a list of navigationController.viewControllers. In case of VC presented modally the mainViewController won't be present on that list. Maybe we could use navigationController.presentedViewController for that check also but it requires more investigation how it behave in case multiple VC are presented modally.
2. Starting new child coordinator from coordinator whose mainVC is presented modally will push/present new VC on UINavigationController that is below current VC presented modally.

Reply:
Presenting the `AboutAppViewController` does show up up in the list though.

```
(lldb) po navigationCoordinatorsTracker.viewControllersToChildCoordinators
▿ 1 element
  ▿ 0 : 2 elements
    ▿ key : <NavigationFlowCoordinatorExample.AboutAppViewController: 0x7fc705a297d0>
    ▿ value : <NavigationFlowCoordinatorExample.AboutAppCoordinator: 0x600000255180>
```
It's why I call `forceUpdateTrackingStatus()` within the `present(...)` function to ensure it gets tracked.  Otherwise I noticed it wasn't being `deinit()`.

Reply:
Ad 1. Agree, the coordinator and its mainViewAppear appears in viewControllersToChildCoordinators map, but its mainViewController is not included in navigationController.viewControllers and this is why the coordinator is deallocated in next cleanUpChildCoordinators() call.

And forceUpdateTrackingStatus() causes rather throwing out unused coordinators than force storing them in  viewControllersToChildCoordinators map